copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopyBookSource.java

```diff
@@ -0,0 +1,283 @@
+      }
+    }
+    emitter.emit(builder.build());
+  }
+
+  /**
+   * Convert each field to data type as specified by user in the output schema
+   *
+   * @param field Schema.Field object defining the field name and properties
+   * @param value value of field to be converted to datatype specified by user
+   * @return value casted to the required data type
+   * @throws IOException
+   */
+  private Object parseValue(Schema.Field field, AbstractFieldValue value) throws IOException {
+    Schema fieldSchema = field.getSchema();
+    Schema.Type fieldType = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema
+      .getType();
+
+    switch (fieldType) {
+      case NULL:
+        return null;
+      case INT:
+        return value.asInt();
+      case DOUBLE:
+        return value.asDouble();
+      case BOOLEAN:
+        return value.asBoolean();
+      case LONG:
+        return value.asLong();
+      case FLOAT:
+        return value.asFloat();
+      case STRING:
+        return value.asString();
+    }
+    throw new IOException(String.format("Unsupported schema: %s for field: \'%s\'",
+                                        field.getSchema(), field.getName()));
+  }
+
+  /**
+   * Create schema from list field names retrieved from CopybookFiles
+   *
+   * @param fieldsSet set of field names extracted from Copybook
+   * @return output schema fields
+   */
+  private Schema getSchema(Set<String> fieldsSet) {
+    List<Schema.Field> fields = Lists.newArrayList();
+    for (String field : fieldsSet) {
+      fields.add(Schema.Field.of(field, Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    }
+    return Schema.recordOf("record", fields);
+  }
+
+  /**
+   * Config class for CopybookSource.
+   */
+  public static class CopybookSourceConfig extends ReferencePluginConfig {
+
+    @Nullable
+    @Description(CopybookConstants.MAX_SPLIT_SIZE_DESCRIPTION)
+    public Long maxSplitSize;
+    @Name("binaryFilePath")
+    @Description("Complete path of the .bin to be read; for example: 'hdfs://10.222.41.31:9000/test/DTAR020_FB.bin' " +
+      "or file:///home/cdap/DTAR020_FB.bin'.\n " +
+      "This will be a fixed-length binary format file that matches the copybook.\n" +
+      "(This is done to accept files present on a remote HDFS location.)")
+    private String binaryFilePath;
+
+    @Name("schema")
+    @Nullable
+    @Description("The schema for the data as it will be formatted in CDAP. Sample schema: " +
+      "{\n" +
+      "    \"type\": \"record\",\n" +
+      "    \"name\": \"schemaBody\",\n" +
+      "    \"fields\": [\n" +
+      "        {\n" +
+      "            \"name\": \"name\",\n" +
+      "            \"type\": \"string\"\n" +
+      "        },\n" +
+      "        {\n" +
+      "            \"name\": \"age\",\n" +
+      "            \"type\": \"int\"\n" +
+      "        }" +
+      "    ]\n" +
+      "}")
+    private String schema;
+
+    @Nullable
+    private Integer fileStructure;
```

Comment:
it does not make any sense to have this property. From the code, there is only one valid option.  If the plugin actually supports multiple file structures, then we can add this. But until then, the only thing this does is confuse people and add another way for somebody to misconfigure this.